# HG changeset patch
# User Rik <rik@octave.org>
# Date 1464883385 25200
#      Thu Jun 02 09:03:05 2016 -0700
# Branch stable
# Node ID 4950810ce6058ff41b5057476210e587fccaef17
# Parent  7720b8246f89759afc3d8f5c910983c1e84c5ac8
__line__.m: Stop intermittent errors when plotting multiple lines (bug #46466).

* __line__.m: Avoid calling cellfun at line 117 if it is known that no
assignment can take place because the LHS index expression is empty.

diff --git a/scripts/plot/draw/private/__line__.m b/scripts/plot/draw/private/__line__.m
--- a/scripts/plot/draw/private/__line__.m
+++ b/scripts/plot/draw/private/__line__.m
@@ -109,18 +109,20 @@ function h = __line__ (p, varargin)
   mask = reshape ([false(1,3); ismat], [1, 6]);
 
   for i = 1:nlines
     tmp = data(ismat);
     if (! size_equal (tmp)
         || (nvecpts != 0 && any (nvecpts != cellfun ("size", tmp, 1))))
       error ("line: data size_mismatch");
     endif
-    data_args(mask) = cellfun (@(x) x(:,i), data(ismat),
-                               "uniformoutput", false);
+    if (any (mask))
+      data_args(mask) = cellfun (@(x) x(:,i), data(ismat),
+                                 "uniformoutput", false);
+    endif
 
     handles(i) = __go_line__ (p, data_args{:}, other_args{:});
 
   endfor
 
   if (nargout > 0)
     h = handles;
   endif
